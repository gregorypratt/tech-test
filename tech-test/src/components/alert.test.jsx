import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Alert from './alert';

describe('Alert', () => {
  it('renders without crashing', () => {
    const component = shallow(<Alert />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('is styled based on type', () => {
    const component = shallow(<Alert type="info" />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('displays the provided message', () => {
    const component = shallow(<Alert message="test message" />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});