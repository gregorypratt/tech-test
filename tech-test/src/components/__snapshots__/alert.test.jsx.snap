// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays the provided message 1`] = `
<div
  className="c-alert c-alert--error"
  role="alert"
>
  test message
</div>
`;

exports[`Alert is styled based on type 1`] = `
<div
  className="c-alert c-alert--info"
  role="alert"
/>
`;

exports[`Alert renders without crashing 1`] = `
<div
  className="c-alert c-alert--error"
  role="alert"
/>
`;
